### PR TITLE
Domains: Add 100-Year Domain specific Thank You page

### PR DESCRIPTION
**Pull request by rafaelgallani**
## Proposed Changes
Adds changes to the existing components so that a specific copy/button is displayed for the 100-year domain thank you page (post-checkout page).

## Why are these changes being made?
These changes are related to the 100-Year Domain UX update project: pcYYhz-2q8-p2.

## Testing Instructions
- Build this branch locally or go to its `calypso.live` link;
- Enable store sandbox so no actual purchase is performed during the tests;
- Go to `/setup/hundred-year-domain` and choose any domain available;
- Complete the checkout and ensure you see the same as in the screenshot below - as opposed to the previous generic "thank you" page.

## Preview
| Before                                                                                    | After                                                                                     |
|-------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------|
| image | image |

## Pre-merge Checklist

- [X] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [X] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/my-sites/checkout/checkout-thank-you/redesign-v2/pages/domain-only.tsx [hunk 1]
```diff
@@ -1,27 +1,87 @@
...
+									#050c16 41.26%,
+									#02080f 88.44%
+								);
+							}
+						` }
+					/>
+					<HundredYearThankYou
+						siteSlug={ String( purchasedDomain.blogId ) }
+						receiptId={ Number( receipt.receiptId ) }
+						productSlug="domain_reg"
```
Comment (gius80) [inline client/my-sites/checkout/checkout-thank-you/redesign-v2/pages/domain-only.tsx]: Maybe we can use `domainProductSlugs.DOTCOM_DOMAIN_REGISTRATION`?

File: client/my-sites/checkout/checkout-thank-you/hundred-year-plan-thank-you/index.tsx [hunk 2]
```diff
@@ -129,7 +135,11 @@ function isSiteCreatedWithinLastHour( createdTime: string ): boolean {
 	return Date.now() - new Date( createdTime ).getTime() < HOUR_IN_MS;
 }
 
-export default function HundredYearPlanThankYou( { siteSlug, receiptId }: Props ) {
+export default function HundredYearThankYou( {
```
Comment (gius80) [inline client/my-sites/checkout/checkout-thank-you/hundred-year-plan-thank-you/index.tsx:138]: I was wondering if we should rename the file to match the classname 🤔

**Review by gius80 — APPROVED** [review]
Tested and works as expected. I left a couple of comments.

**Review by gius80 — APPROVED** [review]
Looks good now 🚀

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/17018598

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @rafaelgallani for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/17018598).

**Comment by sirbrillig** [comment]
I believe this is causing a fatal error on the thank-you page after purchasing a domain. 

Screenshot 2024-12-03 at 12 46 52 PM

The reason is that this type is not actually TypeScript:

https://github.com/Automattic/wp-calypso/blob/03adc44709b14e4a98f2c6a0ad33ecbb1cb0b643/client/state/sites/domains/selectors.js#L9

Since apparently at some point the type definition for `ResponseDomain` moved, it means that this type is invalid and TypeScript thinks that `getDomainsBySiteId()` returns `any[]`. It therefore doesn't notice that `domain` in the `DomainOnlyThankYou` component can be undefined.

**Pull request merged**